src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -309,20 +309,29 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
             await Task.FromResult(true);
         }
 
-        private static async Task HandleSignatureHelpRequest(
+        private static async Task HandleDocumentHighlightRequest(
             TextDocumentPosition textDocumentPosition,
-            RequestContext<SignatureHelp> requestContext)
+            RequestContext<DocumentHighlight[]> requestContext)
         {
             await Task.FromResult(true);
         }
+#endif
 
-        private static async Task HandleDocumentHighlightRequest(
+        private static async Task HandleSignatureHelpRequest(
             TextDocumentPosition textDocumentPosition,
-            RequestContext<DocumentHighlight[]> requestContext)
+            RequestContext<SignatureHelp> requestContext)
         {
-            await Task.FromResult(true);
+            var scriptFile = WorkspaceService<SqlToolsSettings>.Instance.Workspace.GetFile(
+                textDocumentPosition.TextDocument.Uri);
+
+            var help = LanguageService.Instance.GetSignatureHelp(textDocumentPosition, scriptFile);
+            if (help != null)
+            {
+                await requestContext.SendResult(help);
+            }
+            
+            await requestContext.SendResult(new SignatureHelp());
```

Comment:
My bad! Yes, you're correct. I must've interpreted the await as a return while I was writing this since they have the same highlighting color...